contracts/TracerPerpetualSwaps.sol

```
@@ -478,12 +478,15 @@ contract TracerPerpetualSwaps is ITracerPerpetualSwaps, Ownable {
         int256 amountToTakeFromInsurance
     ) external override onlyLiquidation {
         address insuranceAddr = address(insuranceContract);
+        require(
+            balances[insuranceAddr].position.quote - amountToTakeFromInsurance >= 0,
+            "TCR: Insurance not funded enough"
+        );
```

should this throw an error or should we instead just empty the insurance fund of everything its got, even if it holds less than the specified amount?

---

I think this should throw an error. In the current state, this error should actually never be called since the function which calls it in `Liquidation.sol` determines how much should be taken from insurance (depending on if there's a sufficient amount). I.e Liquidation contract is responsible for handling any insufficient funds not the tracer contract which is just there to update balances.